pkg/apis/kubeone/types.go

```diff
@@ -395,8 +399,27 @@ type ClusterNetworkConfig struct {
 
 	// KubeProxy config
 	KubeProxy *KubeProxyConfig `json:"kubeProxy,omitempty"`
+
+	// IPFamily "" | IPv4 | IPv6 | IPv4+IPv6.
+	IPFamily IPFamily `json:"ipFamily,omitempty"`
+
+	// NodeCIDRMaskSizeIPv4
+	NodeCIDRMaskSizeIPv4 *int `json:"nodeCIDRMaskSizeIPv4,omitempty"`
+
+	// NodeCIDRMaskSizeIPv6
+	NodeCIDRMaskSizeIPv6 *int `json:"nodeCIDRMaskSizeIPv6,omitempty"`
 }
 
+// IPFamily IPv4 | IPv6 | IPv4+IPv6.
+type IPFamily string
+
+const (
+	Unspecified IPFamily = "" // interpreted as IPv4
```

Comment:
I think this case shouldn't be possible. We should default to `IPv4` and get rid of this.